fiscal_epos_print/static/src/js/epson_epos_print.js

```diff
@@ -0,0 +1,783 @@
+                break;
+            case "1":
+                receipt = false;
+                // Receipt = "Fiscale/Non fiscale chiuso";
+                break;
+            case "2":
+                receipt = _t("Non fiscal open");
+                break;
+            case "3":
+                receipt = _t("Payment in progress");
+                break;
+            case "4":
+                receipt = _t(
+                    "Error on last ESC/POS command with Fiscal/Non fiscal closed"
+                );
+                break;
+            case "5":
+                receipt = _t("Negative receipt");
+                break;
+            case "6":
+                receipt = _t("Error on last ESC/POS command with Non fiscal open");
+                break;
+            case "7":
+                receipt = _t("Waiting for receipt closing in JAVAPOS mode");
+                break;
+            case "8":
+                receipt = _t("Fiscal document open");
+                break;
+            case "A":
+                receipt = _t("Title open");
+                break;
+            case "B":
+                receipt = _t("Title closed");
+                break;
+            default:
+                receipt = _t("Wrong answer");
+        }
+
+        return printer || ej || receipt;
+    }
+
+    function getStatusField(tag) {
+        return tag === "printerStatus" || tag === "fsStatus";
+    }
+
+    var eposDriver = core.Class.extend({
+        init: function (options, sender) {
+            var self = this;
+            var opts = options || {};
+            this.url = opts.url || "http://192.168.1.1/cgi-bin/fpmate.cgi";
+            // eslint-disable-next-line
+            this.fiscalPrinter = new epson.fiscalPrint();
+            this.sender = sender;
+            this.order = options.order || null;
+            // eslint-disable-next-line
+            this.fiscalPrinter.onreceive = function (res, tag_list_names, add_info) {
+                // TODO not exist
+                // sender.chrome.loading_hide();
+                var tagStatus = tag_list_names
+                    ? tag_list_names.filter(getStatusField)
+                    : [];
+                var msgPrinter = "";
+                var info = "";
+
+                if (tagStatus.length > 0 && res.success) {
+                    info = add_info[tagStatus[0]];
+                    res.success = !isErrorStatus(info);
+                }
+
+                var order = "";
+                if (!res.success) {
+                    if (self.order !== null) {
+                        order = self.order;
+                        order.fiscal_printer_debug_info =
+                            JSON.stringify(res) +
+                            "\n" +
+                            JSON.stringify(tag_list_names) +
+                            "\n" +
+                            JSON.stringify(add_info);
+                        // TODO is push_orders or push_single_order
```

Comment:
AFAIK push_single_order è più corretto in quanto esegue il flush soltanto di questo ordine particolare, mentre push_orders può anche essere eseguito passando un singolo ordine ma questo viene eseguito al localStorage degli ordini locali che poi vengono sincronizzati in toto.

Dai test che abbiamo eseguito è piuttosto importante che venga ripristinata l'istruzione di push a questo punto altrimenti al termine di ogni emissione scontrino sul POS (dalla seconda in avanti, non alla prima) viene visualizzato il popup con conferma "Ordini non sincronizzati residui" che richiede intervento manuale dell'operatore (vedi il blocco finally all'interno di _finalizeValidation() in PaymentScreen.js di point_of_sale)

Reply:
aggiornato la PR
@are-agilebg puoi priprovare a fare quel test?